### PR TITLE
Handle Livy "waiting" statement state

### LINKED ISSUE #317: Handle "waiting" status form Livy
Hello, 

Bumped into an issue with the latest Livy version:

- Livy version 0.3 (built from master)
- SparkMagic version 0.9.3

In **livyclintlib/command.py**, when checking the statement status:
`if status == u"running":`

Currently, the "else" clause assumes that there is an output, which is not the case of "waiting" state, thus throwing an error.

Shouldn't "waiting" state be treated just like "running" state?

### LINKED ISSUE #320: Error creating session: 'NoneType' object is not subscriptable
I updated Jupyterhub and Sparkmagic and i apper this error 

2017-01-11 14:13:34,967 ERROR   SparkMagics     Error creating session: 'NoneType' object is not subscriptable

Complete log:

2017-01-11 14:11:53,751 DEBUG   SparkMagics     Initialized spark magics.
2017-01-11 14:11:53,752 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookLoaded,Timestamp: 2017-01-11 13:11:53.752289
2017-01-11 14:11:53,753 DEBUG   scala_jupyter_kernel    Loaded magics.
2017-01-11 14:11:53,754 DEBUG   scala_jupyter_kernel    Changed language.
2017-01-11 14:11:54,592 DEBUG   scala_jupyter_kernel    Registered auto viz.
2017-01-11 14:11:54,612 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionStart,Timestamp: 2017-01-11 13:11:54.612522,MagicName: help,LivyKind: spark,MagicGuid: cbfb8cd1-edf0-45ba-b539-dd1ad4467c0e
2017-01-11 14:11:54,620 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionEnd,Timestamp: 2017-01-11 13:11:54.620836,MagicName: help,LivyKind: spark,MagicGuid: cbfb8cd1-edf0-45ba-b539-dd1ad4467c0e,Success: True,ExceptionType: ,ExceptionMessage: 
2017-01-11 14:11:54,664 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionStart,Timestamp: 2017-01-11 13:11:54.664805,MagicName: info,LivyKind: spark,MagicGuid: 6656b866-8608-4cb3-91c4-9a838d4b6820
2017-01-11 14:11:54,704 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionEnd,Timestamp: 2017-01-11 13:11:54.704429,MagicName: info,LivyKind: spark,MagicGuid: 6656b866-8608-4cb3-91c4-9a838d4b6820,Success: True,ExceptionType: ,ExceptionMessage: 
2017-01-11 14:11:54,709 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionStart,Timestamp: 2017-01-11 13:11:54.709288,MagicName: configure,LivyKind: spark,MagicGuid: 957fe56c-31db-4b4e-aa00-8cedf4294e0e
2017-01-11 14:11:54,709 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionStart,Timestamp: 2017-01-11 13:11:54.709750,MagicName: info,LivyKind: spark,MagicGuid: 23a0bf4a-d7c9-4613-a065-53eda79442df
2017-01-11 14:11:54,734 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionEnd,Timestamp: 2017-01-11 13:11:54.734331,MagicName: info,LivyKind: spark,MagicGuid: 23a0bf4a-d7c9-4613-a065-53eda79442df,Success: True,ExceptionType: ,ExceptionMessage: 
2017-01-11 14:11:54,734 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookMagicExecutionEnd,Timestamp: 2017-01-11 13:11:54.734537,MagicName: configure,LivyKind: spark,MagicGuid: 957fe56c-31db-4b4e-aa00-8cedf4294e0e,Success: True,ExceptionType: ,ExceptionMessage: 
2017-01-11 14:11:54,740 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookSessionCreationStart,Timestamp: 2017-01-11 13:11:54.740621,SessionGuid: edadabf5-6eab-46b1-b95d-fa518bb44973,LivyKind: spark
2017-01-11 14:11:54,766 DEBUG   LivySession     Session 30 in state starting. Sleeping 100 seconds.
2017-01-11 14:13:34,859 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookStatementExecutionStart,Timestamp: 2017-01-11 13:13:34.859843,SessionGuid: edadabf5-6eab-46b1-b95d-fa518bb44973,LivyKind: spark,SessionId: 30,StatementGuid: 9f55f588-fa62-4f68-8a41-3f048fedbd5c
2017-01-11 14:13:34,966 DEBUG   Command Status of statement 0 is waiting.
2017-01-11 14:13:34,967 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookStatementExecutionEnd,Timestamp: 2017-01-11 13:13:34.966948,SessionGuid: edadabf5-6eab-46b1-b95d-fa518bb44973,LivyKind: spark,SessionId: 30,StatementGuid: 9f55f588-fa62-4f68-8a41-3f048fedbd5c,StatementId: 0,Success: False,ExceptionType: TypeError,ExceptionMessage: 'NoneType' object is not subscriptable
2017-01-11 14:13:34,967 INFO    EventsHandler   InstanceId: edf043f5-2b00-4c3d-b48a-8a1e84070eeb,EventName: notebookSessionCreationEnd,Timestamp: 2017-01-11 13:13:34.967089,SessionGuid: edadabf5-6eab-46b1-b95d-fa518bb44973,LivyKind: spark,SessionId: 30,Status: idle,Success: False,ExceptionType: TypeError,ExceptionMessage: 'NoneType' object is not subscriptable
2017-01-11 14:13:34,967 ERROR   SparkMagics     Error creating session: 'NoneType' object is not subscriptable

### PR DESCRIPTION
**Pull request by aggFTW**
Fixes #320
Fixes #317

### CODE REVIEW

**Comment by aggFTW** [comment]
@1shibol @roblax @prabhu1984

File: sparkmagic/sparkmagic/livyclientlib/command.py [hunk 1]
```diff
@@ -44,24 +44,25 @@ def execute(self, session):
             return output
 
     def _get_statement_output(self, session, statement_id):
-        statement_running = True
         out = u""
```
Comment (msftristew) [inline sparkmagic/sparkmagic/livyclientlib/command.py]: The logic of this is confusing to me.  `out` is never assigned to in this file so it is always `u""`.  What purpose does it serve?

Comment (aggFTW) [inline sparkmagic/sparkmagic/livyclientlib/command.py]: Yeap. Removed the out variable.

**Comment by msftristew** [comment]
On line 70 of `command.py` there is one more reference to the `out` variable, please remove it

**Comment by tc0312** [comment]
LGTM

**Pull request merged**